### DISCUSSION TITLE
Fastify 4.x- serializing response body in onResponse hook

### DISCUSSION THREAD

**Discussion by rhodee** [fastify/fastify#5650] [Q&A] [answered]
Greetings, 

What is the best current practice for this use case?

I am building a handler to serialize responses as a string after the response is sent.

I tried converting the reply.raw stream to a string but ran into circular ref errors. Other hooks provide the payload as an arg. 

Thank you.

**Comment by mcollina** [comment, marked as answer]
It's not possible. You'd need to intercept the payload before the response is sent.

**Reply by rhodee** [reply]
The docs suggest the onResponse hook is a good place for handling my use case, which includes recording a response scoped to an individual request:

> The onResponse hook is executed when a response has been sent, so you will not be able to send more data to the client. It can however be useful for sending data to external services, for example, to gather statistics.

I like the idea of doing this after a response is sent. It would be great if this could be supported. Thank you.
